pkg/services/comments/commentmodel/permissions.go

```diff
@@ -104,6 +105,12 @@ func (c *PermissionChecker) CheckWritePermissions(ctx context.Context, orgId int
 		if !c.features.IsEnabled(featuremgmt.FlagAnnotationComments) {
 			return false, nil
 		}
+		if c.features.IsEnabled(featuremgmt.FlagAccesscontrol) {
+			evaluator := accesscontrol.EvalPermission(accesscontrol.ActionAnnotationsWrite, accesscontrol.ScopeAnnotationsTypeDashboard)
+			if canEdit, err := c.accessControl.Evaluate(ctx, signedInUser, evaluator); err != nil || !canEdit {
+				return canEdit, err
+			}
+		}
```

Comment:
ah, this is because of the previous change we did, where we now need to include the `user` ? 🤔

Reply:
No, this bit is to ensure that only users who are allowed to edit annotations can add comments to them.